webpack.config.extensions.js

```diff
@@ -36,112 +28,50 @@ function blockScripts( type, inputDir, presetBlocks ) {
+	if ( fs.existsSync( viewScriptPath ) ) {
+		viewBlocks[ block + '/view' ] = [ viewSetup, ...[ viewScriptPath ] ];
+	}
+	return viewBlocks;
+}, {} );
 
-	// Combines all the different blocks into one editor-beta.js script
-	const editorBetaScript = [
-		editorSetup,
-		...blockScripts( 'editor', path.join( __dirname, 'extensions' ), allPresetBlocks ),
-	];
+// Combines all the different blocks into one editor.js script
+const editorScript = [
+	editorSetup,
+	...blockScripts( 'editor', path.join( __dirname, 'extensions' ), presetBlocks ),
+];
 
-	const webpackConfig = {
-		bail: ! isDevelopment,
-		context: __dirname,
-		entry: {
-			editor: editorScript,
-			'editor-beta': editorBetaScript,
-			...viewBlocksScripts,
-		},
-		mode: isDevelopment ? 'development' : 'production',
-		devtool: process.env.SOURCEMAP || ( isDevelopment ? '#eval' : false ),
-		output: {
-			path: path.join( __dirname, '_inc', 'blocks' ),
-			filename: '[name].js',
-			libraryTarget: 'window',
-		},
-		optimization: {
-			minimize: ! isDevelopment,
-			minimizer: Minify( {
-				cache: process.env.CIRCLECI
-					? `${ process.env.HOME }/terser-cache`
-					: 'docker' !== process.env.CONTAINER,
-				parallel: workerCount,
-				sourceMap: Boolean( process.env.SOURCEMAP ),
-				terserOptions: {
-					ecma: 5,
-					safari10: true,
-					mangle: true,
-				},
-			} ),
-		},
-		module: {
-			rules: [
-				TranspileConfig.loader( {
-					workerCount,
-					configFile: path.join( __dirname, 'babel.config.js' ),
-					cacheDirectory: true,
-					exclude: /node_modules\//,
-				} ),
-				SassConfig.loader( {
-					preserveCssCustomProperties: false,
-					includePaths: [ path.join( __dirname, 'client' ) ],
-					prelude: '@import "~@automattic/calypso-color-schemes/src/shared/colors";',
-				} ),
-				FileConfig.loader(),
-			],
-		},
-		resolve: {
-			extensions: [ '.json', '.js', '.jsx' ],
-			modules: [ 'node_modules' ],
-		},
-		node: false,
-		plugins: [
-			new webpack.DefinePlugin( {
-				'process.env.NODE_ENV': JSON.stringify( process.env.NODE_ENV ),
-				global: 'window',
-			} ),
-			new webpack.IgnorePlugin( /^\.\/locale$/, /moment$/ ),
-			...SassConfig.plugins( { cssFilename, minify: ! isDevelopment } ),
-			new DuplicatePackageCheckerPlugin(),
-			new CopyWebpackPlugin( [
-				{
-					from: presetPath,
-					to: 'index.json',
-				},
-			] ),
-		],
-		externals: [ wordpressExternals, 'wp', 'lodash' ],
-	};
+// Combines all the different blocks into one editor-beta.js script
+const editorBetaScript = [
+	editorSetup,
+	...blockScripts( 'editor', path.join( __dirname, 'extensions' ), allPresetBlocks ),
+];
 
-	return webpackConfig;
-}
+const webpackConfig = getBaseWebpackConfig( null, {
+	entry: {
+		editor: editorScript,
+		'editor-beta': editorBetaScript,
+		...viewBlocksScripts,
+	},
+	'output-path': path.join( __dirname, '_inc', 'blocks' ),
+} );
 
-module.exports = getWebpackConfig;
+module.exports = _.merge( {}, webpackConfig, {
```

Comment:
Generally good style, but we've just produced the webpackConfig, we "own" it. Also, merging into a new object is slightly misleading because `merge` is deep, so the object contained in `webpackConfig` will be mutated anyways.

We could merge right where we create the config: `module.exports = _.merge( getBaseWebpackConfig(…), { … } )`

```suggestion
module.exports = _.merge( webpackConfig, {
```

Reply:
> Generally good style, but we've just produced the webpackConfig, we "own" it. Also, merging into a new object is slightly misleading because `merge` is deep, so the object contained in `webpackConfig` will be mutated anyways.

:thinking: This has me confused. I'm reading 'deep' merge as 'recursively merging properties from source objects into the destination one' but I didn't expect source objects to be mutated? See https://lodash.com/docs/4.17.11#merge

> We could merge right where we create the config: `module.exports = _.merge( getBaseWebpackConfig(…), { … } )`

Yeah, that makes sense :+1: I was avoiding it because of nesting levels, but it shouldn't be too bad since there aren't too many LOCs involved here.

Reply:
Hmm it looks a bit more distracting with the `getBaseWebpackConfig()` call inlined after all :sweat_smile:

Reply:
Apparently my interpretation was off, it _doesn't_ seem to mutate the contents:

```js
const { merge } = require('lodash')
const a = { b: { orig: 'b' } }
const toMerge = { b: { new: 'b' } }
const a_ = merge( {}, a, toMerge )
a.b  // { orig: 'b' }
a_.b // { orig: 'b', new: 'b' }
```

Reply:
Please, leave this. It's good as-is 👍